File: Functional Programming/Haskell/03_Aquecimento/min3.hs
min3 a b c
    | a < b && b <= c = a
    | b < a && a <= c = b
    | c < a && a <= b = c
    | otherwise = a
    
{- 
	--IN : Três números, x, y e z
	--OUT: Menor valor entre x, y e z
	
	min3 1 2 3 == 1
	min3 2 1 3 == 1
	min3 3 4 2 == 2
	min3 2 5 4 == 2
-}

File: Functional Programming/Python-Funcional/03. Aquecimento/min3.py
# retorna o menor valor entre três números
def min3(a, b, c):
    if a < b and b <= c:
        return a
    elif b < a and a <= c:
        return b
    elif c < a and a <= b:
        return c
    else:
        return a

print(min3(1, 2, 3))
print(min3(2, 1, 3))
print(min3(3, 4, 2))
print(min3(2, 5, 4))